### ISSUE TITLE
Non-reserved keywords are converted/rewritten to lowercase

### ISSUE THREAD

**Issue by gedgar** [vitessio/vitess#5821]
#### Overview of the Issue

Non-reserved keywords work in Vitess:
```
/*
  These are non-reserved Vitess, because they don't cause conflicts in the grammar.
  Some of them may be reserved in MySQL. The good news is we backtick quote them
  when we rewrite the query, so no issue should arise.
  Sorted alphabetically
*/
```
We convert them to lowercase, 
VTgate:
```
mysql> select Status,Names from test;
+--------+-------+
| status | names |
+--------+-------+
```
While MySQL doesn't:
```
mysql> select Status,Names from test;
+--------+-------+
| Status | Names |
+--------+-------+
```

#### Binary version
```sh
Version: 30b2ff57b (Git branch 'master') built on Fri Feb 14 08:48:35 PST 2020 by gary@gary-ThinkPad using go1.13.4 linux/amd64
```

**Comment by deepthi** [comment]
What problem does this cause?

**Comment by gedgar** [comment]
Fixed by #5897
